### ISSUE TITLE
About DST equation 7 （Debiased Self-Training for Semi-Supervised Learning）

### ISSUE THREAD

**Issue by gufan-d** [Ledzy/MDS5210#1]
Sorry to bother you.I'm new for Semi-Supervised Learning.

I find it difficult to correspond formula 7 in paper with the code.
minψLU(ψ,hworst(ψ), ̂fψ,h) −LL(ψ,hworst(ψ)).
```
    def forward(self, y_l, y_l_adv, y_u, y_u_adv):
        _, prediction_l = y_l.max(dim=1)
        loss_l = self.eta_prime * F.cross_entropy(y_l_adv, prediction_l)

        _, prediction_u = y_u.max(dim=1)
        loss_u = F.nll_loss(shift_log(1. - F.softmax(y_u_adv, dim=1)), prediction_u)

        return loss_l + loss_u
```
Why do we write it[1. - F.softmax(y_u_adv, dim=1)] this way?
I hope to get your answer,thank you!

**Comment by Ledzy** [comment]
Hey,

This repo is forked from https://github.com/thuml/Transfer-Learning-Library. Please leave your issue here.

**Comment by gufan-d** [comment]
I'm sorry.
